test/cppsuite/test_harness/workload_generator.h

```diff
@@ -40,6 +41,37 @@
 #include "workload/workload_tracking.h"
 
 namespace test_harness {
+/*
+ * Helper class to enable scalable operation types in the database_operation. In order to add a new
+ * standard operation function one need only add a new type, the equivalent function to the database
+ * operation class, and update the get_func function.
+ */
+class operation_config {
+    public:
+    /* FIXME-T-F: Should we just use an emum for type? */
+    operation_config(configuration *config, std::string type)
+        : config(config), type(type), thread_count(config->get_int(THREAD_COUNT))
+    {
+    }
+
+    /* Returns a function pointer to the member function of the supplied database operation. */
+    std::function<void(test_harness::thread_context *)>
+    get_func(database_operation *dbo)
+    {
+        if (type == "insert")
+            return std::bind(&database_operation::insert_operation, dbo, std::placeholders::_1);
+        else if (type == "read")
+            return std::bind(&database_operation::read_operation, dbo, std::placeholders::_1);
+        else if (type == "update")
+            return std::bind(&database_operation::update_operation, dbo, std::placeholders::_1);
+        else
+            testutil_die(-1, "invalid thread type");
```

Comment:
EINVAL instead of -1 and you can add the type value in the error message.

Reply:
f6b340f